### PR TITLE
refactor travis setup

### PR DESCRIPTION
**Pull request by jsvd**
rely on config imports to maximize reuse. For a plugin to adopt these scripts they only need to either add a simple:

```yml
import:
- logstash-plugins/.ci:travis/defaults.yml@travis_refactor
- logstash-plugins/.ci:travis/matrix.yml@travis_refactor
- logstash-plugins/.ci:travis/exec.yml@travis_refactor
```

This pull request tries to do away with submodules to reduce the effort of adoption. The [exec.yml](https://github.com/logstash-plugins/.ci/blob/travis_refactor/travis/exec.yml#L1) downloads a tarball with all the necessary docker and shell scripts, but skips any existing files, so plugins can customize testing.

The goal is to version the yaml and scripts through git tags, so all plugins rely on the same setup (unless it is customized).

### CODE REVIEW

**Comment by jsvd** [comment]
I have updated the PR to reduce the scope (drop auto publishing).
Here are a few examples of adopting these configurations:
- http output: https://github.com/logstash-plugins/logstash-output-http/pull/108
- date filter: https://github.com/logstash-plugins/logstash-filter-date/pull/135 
- udp input: https://github.com/logstash-plugins/logstash-input-udp/pull/49

File: travis/defaults.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+os: linux
+language: ruby
```
Comment (kares) [inline travis/defaults.yml]: out of curiosity, is there anything we might need the system installed ruby for?

Comment (jsvd) [reply travis/defaults.yml]: not really, the travis config parser complains about us not setting any lang, we can also set this as `generic`

File: travis/exec.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+before_install:
+  - mkdir -p .ci && curl -sL https://github.com/logstash-plugins/.ci/archive/travis_refactor.tar.gz | tar zxvf - --skip-old-files --strip-components=1 -C .ci --wildcards "*Dockerfile*" "*docker*" "*.sh"
```
Comment (kares) [inline travis/exec.yml]: the decision left is whether we point everything to master or start with plugins importing `travis.yml` from a tag.
master should be fine as a starter (we might still want to tag `v0.1.0` or whatever and sync all places to use tag):
- travis/exec.yml `curl -sL https://github.com/logstash-plugins/.ci/archive/{master|v0.1.0}.tar.gz`
- travis/travis.yml `logstash-plugins/.ci:travis/defaults.yml@{master|v0.1.0}`
- README to rather mention master (or a tag) e.g. 
```yaml
import:
- logstash-plugins/.ci:travis/defaults.yml@master
# ...
```

File: travis/defaults.yml (hunk 1 above)
Comment (kares) [reply travis/defaults.yml]: there's also `language: minimal` (has Docker available) which we used before

File: travis/exec.yml (hunk 2 above)
Comment (jsvd) [reply travis/exec.yml]: ++ my plan is to merge this into master and then push a v0.1.0 commit + tag

**Pull request merged**